apps/google-analytics-4/frontend/src/components/main-app/ChartContent/ChartContent.tsx

```diff
@@ -21,16 +24,26 @@ const ChartContent = (props: Props) => {
     });
   };
 
-  return (
-    <>
+  const renderChartContent = () => {
+    if (error) {
+      return <Note body={error?.message || DEFAULT_ERR_MSG} variant="negative" />;
+    }
+
+    if (pageViewData.rows && !pageViewData.rows.length) {
```

Comment:
I think that this needs to be based on `rowCount` instead of `rows`, perhaps? The old `renderChartContent` function in the `AnalyticsApp` used `rowCount` to determine whether to render the chart content. What I am seeing locally is that if run_report returns no data (rowCount is 0, but there are still some rows, all of which have 0 as the value), then I am still seeing the sidebar chart. I shouldn't be seeing the sidebar chart if there is no data, right?

Reply:
great catch! This was a tricky one as my solution here was to originally solve a race condition that was occurring of rendering, but I believe the refactor of components ultimately solved this, so I am able to revert back to `rowCount`, which as point out more accurately reflects the data!